src/main/java/org/json/JSONArray.java

```diff
@@ -1338,7 +1349,27 @@ public JSONArray put(int index, long value) throws JSONException {
      *             If a key in the map is <code>null</code>
      */
     public JSONArray put(int index, Map<?, ?> value) throws JSONException {
-        this.put(index, new JSONObject(value));
+        this.put(index, new JSONObject(value, new JSONParserConfiguration()));
+        return this;
+    }
+
+    /**
+     * Put a value in the JSONArray, where the value will be a JSONObject that
+     * is produced from a Map.
+     *
+     * @param index
+     *          The subscript
+     * @param value
+     *          The Map value.
+     * @param jsonParserConfiguration
+     *          Configuration for recursive depth
```

Comment:
Please just call this a configuration object. The things that it can configure will change going forward